src/cli/trn_cli_common.c

```diff
@@ -151,6 +153,27 @@ int trn_cli_parse_net(const cJSON *jsonobj, struct rpc_trn_network_t *net)
 		}
 	}
 	net->switches_ips.switches_ips_len = i;
+	// Cluster gateway is NULL or empty if it is not set in a config map which is a string value
+	net->cluster_gateway = 0;
+	if (cluster_gateway != NULL && cJSON_IsString(cluster_gateway) && strlen(cluster_gateway->valuestring) > 0) {
+		struct sockaddr_in sa;
+		inet_pton(AF_INET, cluster_gateway->valuestring, &(sa.sin_addr));
+		net->cluster_gateway = sa.sin_addr.s_addr;
+	}
+        // Virtual can be NULL if it is a local subnet. 0 as false and 1 as true
+	net->virtual = 0;
+	if (virtual != NULL) {
+		if (cJSON_IsBool(virtual)) {
+			if (cJSON_IsFalse(virtual) == 0) {
+				net->virtual = 1;
+			} else {
+				net->virtual = 0;
+			}
```

Comment:
I feel this if block starting on line 166 can be replaced by 

```c
if (cJSON_IsBool(virtual) && cJSON_IsFalse(virtual) == 0) {
     net->virtual = 1;
} 
```

Reply:
We have to check if the virtual is NULL since the incoming json request might not have virtual fields.

Reply:
understand. I'm just trying to remove the most inner if block.

net->virtual is already set to 0 on line 164.

so the if block starting on line 166 (not 165) only needs to set net->virtual to 1 if  "cJSON_IsBool(virtual) && cJSON_IsFalse(virtual) == 0" right?

Reply:
oh but the print_err on line 173 might need to be re-positioned. 

see if it's possible to reduce this section from 3-level if statement to at least 2-level as an optimization.

Reply:
Updated